### PR TITLE
Starting on UI

### PR DESCRIPTION
**Pull request by YoungSong99**
Updated features:
- add bootstrap nav bar, alert and notice
- update sample data (alice, bob)
- force user sign in

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
```
Comment (JoseMDO) [inline app/controllers/application_controller.rb:2]: Nice!

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -6,21 +6,23 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
+    <%= render "shared/cdn_assets" %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:9]: Great job adding your CDN assets to a partial! This helps keep the layout file clean :)

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -6,21 +6,23 @@
...
     <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>
     <%= javascript_importmap_tags %>
-    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-QWTKZyjpPEjISv5WaRU9OFeRpok6YctnYmDr5pNlyT2bRjXh0JMhjY6hW+ALEwIH" crossorigin="anonymous">
-    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/all.min.css">
 
   </head>
-
-  <%= render 'layouts/nav' %>
+  <%= render 'shared/nav' %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:14]: same here

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -6,21 +6,23 @@
...
   <body>
-    <p class="notice"><%= notice %></p>
-    <p class="alert"><%= alert %></p>
     <div class="container">
+      <% if notice.present? %>
+        <%= render "shared/flash", message: notice, css_class: "success" %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:18]: same here

File: app/views/shared/_nav.html.erb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      <span class="navbar-toggler-icon"></span>
+    </button>
+    <div class="collapse navbar-collapse" id="navbarSupportedContent">
+      <% if user_signed_in? %>
+        <ul class="navbar-nav me-auto">
+          <li class="nav-item">
+            <a class="nav-link" href="/photos">Photos</a>
+          </li>
+          <li class="nav-item">
+            <a class="nav-link" href="/comments">Comments</a>
```
Comment (JoseMDO) [inline app/views/shared/_nav.html.erb:15]: remember we can start to use helper methods now like link_to :)

File: lib/tasks/dev.rake [hunk 6]
```diff
@@ -11,6 +11,20 @@ task sample_data: :environment do
     User.destroy_all
   end
 
+  usernames = Array.new { Faker::Name.first_name }
+
+  usernames << "alice"
+  usernames << "bob"
+
+  usernames.each do |username|
```
Comment (JoseMDO) [inline lib/tasks/dev.rake:19]: great job on sample data!

**Review by JoseMDO — APPROVED** [review]
Great job Young!

**Review by ghost — APPROVED** [review]
Excellent work!

**Pull request merged**